src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4517,46 +4517,71 @@ void *wm_vuldet_main(wm_vuldet_t * vuldet) {
 int wm_vuldet_set_agents_info(agent_software **agents_software, update_node **updates) {
     agent_software *agents = NULL;
     agent_software *f_agent = NULL;
-    char global_db[OS_FLSIZE + 1];
-    sqlite3 *db;
-    sqlite3_stmt *stmt = NULL;
     int dist_error;
-    char *id;
-    char *name;
-    char *register_ip;
-    char *os_name;
-    char *os_version;
-    char *arch;
-    int build;
+    char str_id[OS_SIZE_128] = "";
+    char *name = NULL;
+    char *register_ip = NULL;
+    char *os_name = NULL;
+    char *os_version = NULL;
+    char *arch = NULL;
+    int build = 0;
+    int i = 0;
+    int id = -1;
     vu_feed agent_dist_ver;
     vu_feed agent_dist;
+    int *id_array = NULL;
+    cJSON *json_agt_info = NULL;
+    cJSON *json_field = NULL;
 
-    snprintf(global_db, OS_FLSIZE, "%s%s/%s", isChroot() ? "/" : "", WDB2_DIR, WDB_GLOB_NAME);
+    id_array = wdb_get_agents_by_keepalive("<", DISCON_TIME);
```

Comment:
I think that this will cause `Vulnerability Detector` to never scan agents. The thing is, you are always looking for agents with a keepalive less than 1800 (because of the definition of `DISCON_TIME`) and this will never happen.
Should this call be something like this?

```c
id_array = wdb_get_agents_by_keepalive("<", time(0) - DISCON_TIME);
```

Reply:
Fixed!